src/app/collective-rewards/leaderboard/BuildersLeaderBoardTable.tsx

```diff
@@ -0,0 +1,422 @@
+      <LazyRewardCell rewards={rewards} />
+    </TableCell>
+  )
+}
+
+const ProjectedRewardCell = ({ rewards }: { rewards: Reward[] }) => {
+  return (
+    <TableCell className={cn(tableHeaders[3].className, 'border-solid')}>
+      <LazyRewardCell rewards={rewards} />
+    </TableCell>
+  )
+}
+
+type ShareProps = {
+  // a percentage without decimals
+  share: bigint
+}
+
+const ShareCell = ({ share }: ShareProps) => {
+  return (
+    <TableCell className={cn(tableHeaders[4].className, 'border-solid text-center border-b-0 items-center')}>
+      <div className="flex flex-row gap-2 items-center">
+        <ProgressBar progress={Number(share)} color="#1bc47d" />
+        <Label>{share.toString()}%</Label>
+      </div>
+    </TableCell>
+  )
+}
+const ActionCell = () => {
+  /* TODO: manage the button status 
+    - disabled when the backer cannot vote on the Builder
+    - variant=primary when the builder is selected and text changed to "Selected"
+    - variant=secondary by default and text is "Select"
+  */
+  /* TODO: add the onClick event
+   *  - it needs to interact with the allocation context to add the builder to the selected builders
+   */
+  return (
+    <TableCell className={cn(tableHeaders[5].className, 'border-solid align-center')}>
+      <Button variant="secondary" /* disabled={true} */>Select</Button>
+    </TableCell>
+  )
+}
+
+enum RewardsColumnKeyEnum {
+  builder = 'builder',
+  lastCycleReward = 'lastCycleReward',
+  projectedReward = 'projectedReward',
+  rewardPercentage = 'rewardPercentage',
+  share = 'share',
+}
+const tableHeaders = [
+  { label: 'Builder', className: 'w-[14%]', key: RewardsColumnKeyEnum.builder },
+  { label: 'Backer Rewards %', className: 'w-[10%]', key: RewardsColumnKeyEnum.rewardPercentage },
+  { label: 'Last Cycle Rewards', className: 'w-[22%]', key: RewardsColumnKeyEnum.lastCycleReward },
+  { label: 'Est. Backers Rewards', className: 'w-[22%]', key: RewardsColumnKeyEnum.projectedReward },
+  {
+    label: 'Total Allocations',
+    className: 'w-[18%]',
+    tooltip: `The Builder's share of the total allocations`,
+    key: RewardsColumnKeyEnum.share,
+  },
+  // TODO: text-center isn't applied
+  { label: 'Actions', className: 'w-[14%] text-center'},
+]
+
+type ISortConfig = {
+  key: RewardsColumnKeyEnum
+  direction: 'asc' | 'desc'
+}
+
+const BuildersLeaderBoardTable = () => {
+  const { data: rewardsData, isLoading, error: rewardsError } = useGetBuildersRewards()
+  const { setMessage: setErrorMessage } = useAlertContext()
+  const [sortConfig, setSortConfig] = useState<ISortConfig>({
+    key: RewardsColumnKeyEnum.builder,
+    direction: 'asc',
+  })
+
+  // pagination
+  const buildersPerPage = 10
+  const tableDataLength = useMemo(() => Object.keys(rewardsData).length, [rewardsData])
+  const maxPages = useMemo(() => Math.ceil(tableDataLength / buildersPerPage), [tableDataLength])
+  const { paginationUi, currentPage } = useBasicPaginationUi(maxPages)
+
+  useEffect(() => {
```

Comment:
Here we already have a method to [handle](https://github.com/RootstockCollective/dao-frontend/blob/develop/src/app/collective-rewards/utils/handleErrors.ts) errors.